libcodechecker/analyze/plist_parser.py

```diff
@@ -3,170 +3,165 @@
+                LOG.debug('Guessed check name: ' + checker_name)
 
-                elif item['kind'] == 'control':
-                    for edge in item['edges']:
-                        start = make_range(edge.start, files)
-                        end = make_range(edge.end, files)
+            hash_value = diag.get('issue_hash_content_of_line_in_context')
 
-                        if start != current.get_last_path():
-                            current.add_to_path(start)
+            new_report = Report(checker_name,
+                                hash_value,
+                                diag['category'],
+                                diag['type'])
 
-                        current.add_to_path(end)
+            # Set the main diagnostic section used at the listing of reports.
+            start_pos = Position(line=diag['location']['line'],
+                                 col=diag['location']['col'],
+                                 filepath=files[diag['location']['file']])
 
-            current.msg = diag['description']
-            current.category = diag['category']
-            current.type = diag['type']
+            main_section = DiagSection(kind="main", msg=message)
+            main_range = Range(start_pos)
+            main_section.start_range = main_range
+            main_section.end_range = main_range
 
-            try:
-                current.checker_name = diag['check_name']
-            except KeyError as kerr:
-                LOG.debug("Check name wasn't found in the plist file. "
-                          "Read the user guide!")
-                current.checker_name = plist_helper.get_check_name(current.msg)
-                LOG.debug('Guessed check name: ' + current.checker_name)
-
-            try:
-                current.hash_value = \
-                    diag['issue_hash_content_of_line_in_context']
-            except KeyError as kerr:
-                # Hash was not found.
-                # Generate some hash for older clang versions.
-                LOG.debug(kerr)
-                LOG.debug("Hash value wasn't found in the plist file. "
-                          "Read the user guide!")
-                current.hash_value = plist_helper.gen_bug_hash(current)
+            new_report.main_section = main_section
+
+            # Build a bug path from the DiagSections.
+            bug_path = BugPath()
+            for item in diag['path']:
+                # Only event and control items are added to the path.
+                if item['kind'] == 'event':
+                    event = __handle_event(item, files)
+                    bug_path.add_diag(event)
+
+                elif item['kind'] == 'control':
+                    ctrl_sections = __handle_control(item, files)
+                    for cs in ctrl_sections:
+                        bug_path.add_diag(cs)
+                else:
+                    LOG.warning('Not supported item kind found: ' +
```

Comment:
This messages feels clunky on the output. Maybe `Unsupported item kind (VALUE) found during plist parsing`?